twine/settings.py

```diff
@@ -340,19 +340,19 @@ def check_repository_url(self) -> None:
                 repository_url, utils.DEFAULT_REPOSITORY, utils.TEST_REPOSITORY
             )
 
-        if repository_url.startswith(
-            ("https://pypi.org/", "https://test.pypi.org/", "https://upload.pypi.org/")
-        ) and repository_url not in [utils.DEFAULT_REPOSITORY, utils.TEST_REPOSITORY]:
+        if "pypi.org" in repository_url and repository_url not in [
+            utils.DEFAULT_REPOSITORY,
+            utils.TEST_REPOSITORY,
+        ]:
             raise exceptions.InvalidPyPIUploadURL(
                 f"The configured repository {repository_url} is not a known "
                 f"PyPI repository.\n"
                 f"Did you mean {utils.DEFAULT_REPOSITORY} "
-                f"or {utils.TEST_REPOSITORY} ?\n"
-                f"Check your --repository-url value or "
-                f"modify the url in {self.config_file} "
-                f"(if you are using -r or --repository),\n"
-                f"so twine can upload your distributions properly.\n"
-                f"See {utils.PYPIRC_DOCS} for more details."
+                f"or {utils.TEST_REPOSITORY}?\n"
```

Comment:
@bhrutledge I think we need to add a little space here?
Otherwise, folks will easily get an URL like this `https://test.pypi.org/legacy/?`